src/cdn-logs-report/utils/query-builder.js

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2025 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/* c8 ignore start */
+import { DEFAULT_COUNTRY_PATTERNS } from '../constants/country-patterns.js';
+import { extractCustomerDomain, loadSql } from './report-utils.js';
+import { DEFAULT_PATTERNS, DOMAIN_SPECIFIC_PATTERNS, FALLBACK_CASE_STATEMENT } from '../constants/page-patterns.js';
+import { getProviderPattern } from '../constants/user-agent-patterns.js';
+
+function buildDateFilter(startDate, endDate) {
+  const formatPart = (date) => ({
+    year: date.getUTCFullYear().toString(),
+    month: (date.getUTCMonth() + 1).toString().padStart(2, '0'),
+    day: date.getUTCDate().toString().padStart(2, '0'),
+  });
+
+  const start = formatPart(startDate);
+  const end = formatPart(endDate);
+
+  return start.year === end.year && start.month === end.month
+    ? `(year = '${start.year}' AND month = '${start.month}' AND day >= '${start.day}' AND day <= '${end.day}')`
+    : `((year = '${start.year}' AND month = '${start.month}' AND day >= '${start.day}')
+       OR (year = '${end.year}' AND month = '${end.month}' AND day <= '${end.day}'))`;
+}
+
+function buildWhereClause(conditions = [], provider = null) {
+  if (provider) {
+    const pattern = getProviderPattern(provider);
+    conditions.push(`REGEXP_LIKE(user_agent, '${pattern}')`);
+  }
+  return conditions.length > 0 ? `WHERE ${conditions.join(' AND ')}` : '';
+}
+
+function buildWeeklyColumns(periods) {
+  return periods.weeks.map((week) => {
+    const weekKey = week.weekLabel.replace(' ', '_').toLowerCase();
+    const dateFilter = buildDateFilter(week.startDate, week.endDate);
+    return `SUM(CASE WHEN ${dateFilter} THEN count ELSE 0 END) as ${weekKey}`;
+  }).join(',\n      ');
+}
+
+function buildOrderBy(periods) {
+  return periods.weeks
+    .map((week) => week.weekLabel.replace(' ', '_').toLowerCase())
+    .join(' + ');
+}
+
+// Page Type Classification
+function generatePageTypeClassification(site) {
+  const domain = site ? extractCustomerDomain(site) : 'default';
+  const patterns = DOMAIN_SPECIFIC_PATTERNS[domain] || DEFAULT_PATTERNS;
```

Comment:
please retrieve the patterns from 

```
site.getConfig()?.getGroupedURLs('cdn-analysis')
```

if this is undefined, we can fallback to `DEFAULT_PATTERNS